darts/models/filtering_model.py

```diff
@@ -0,0 +1,108 @@
+"""
+Filtering Model Base Class
+------------------------------
+
+A filtering model captures the measured values of a time series as a function of the past as follows:
+
+.. math:: y_{t+1} = f(y_{t+1}, y_t, y_{t-1}, ..., y_1),
```

Comment:
This is the same definition as the forecasting model.
What do we want these "smoothing models" to do? In my understanding they should rather do something like 
constructing a new series `z` of the same length as `y`, where each element `z_t` is a function of the whole series `y`, no?

Reply:
There is a little change in indexing of the formula. Imho any filter/smoothing function looks at current and past values in general. Thats why i said t+1=t+1. I will add more description to avoid confusion with predicting formula